<?xml version="1.0" ?>
<launch>

  <arg name="file" default="camera" />

  
  <include file="$(find openni2_launch)/launch/openni2.launch" >
    <arg name="publish_tf" value="false" />
    <arg name="camera" value="camera" />
    <arg name="depth_registration" default="true" />
    <!--arg name="device_id" value="3@1"/-->
    <arg name="rgb_camera_info_url"   value="file://$(find rgbd_depth_correction)/yaml/kinect_rgb_PS1080_PrimeSense.yaml" />
    <arg name="depth_camera_info_url" value="file://$(find rgbd_depth_correction)/yaml/kinect_depth_PS1080_PrimeSense.yaml" />
  </include>


  <node pkg="target_finder" type="target_locator_srv" name="target_locator_service" output="screen" >

    <!-- Be sure to use RGB projection matrix parameters here after performing intrinsic calibration -->
    <rosparam>
      image_topic: "/camera/rgb/image_rect_color"
      target_type: 2
      target_rows: 5
      target_cols: 5
      target_circle_dia: 0.05
      target_spacing: 0.07
      focal_length_x: 535.023742675781
      focal_length_y: 538.773559570312
      center_x: 305.118641972633
      center_y: 237.842615398564
    </rosparam>
  </node>

  <node pkg="rgbd_depth_correction" type="depth_calibration" name="depth_cal_node" output="screen" >
    <remap from="depth_points" to="/camera/depth_registered/points" />
    <remap from="image" to="/camera/rgb/image_rect_color" />

    <param name="filepath" value="$(find rgbd_depth_correction)/yaml" />
    <param name="filename" value="$(arg file)" />

    <rosparam>
      target_pos_x: 0.0
      target_pos_y: 0.0
      target_pos_z: 0.6
      target_orn_x: 1.0
      target_orn_y: 0.0
      target_orn_z: 0.0
      target_orn_w: 0.0

    </rosparam>
  </node>

  <node name="image_view" pkg="image_view" type="image_view">
    <remap from="image" to="/camera/rgb/image_rect_color" />
  </node>
</launch>
